### PR TITLE
Fixed setting cookies to an external server.

### PR DESCRIPTION
**Pull request by dkorolev**

### CODE REVIEW

File: Blocks/HTTP/impl/posix_client.h [hunk 1]
```diff
@@ -81,8 +81,15 @@ class HTTPClientPOSIX final {
       for (const auto& h : request_headers_) {
         connection.BlockingWrite(h.header + ": " + h.value + "\r\n", true);
       }
-      for (const auto& h : request_headers_.cookies) {
-        connection.BlockingWrite("Set-Cookie: " + h.first + "=" + h.second + "\r\n", true);
+      if (!request_headers_.cookies.empty()) {
```
Comment (mzhurovich) [inline Blocks/HTTP/impl/posix_client.h]: Don't you want to make it a method `CookiesAsString()` in `Headers`?

Comment (mzhurovich) [inline Blocks/HTTP/impl/posix_client.h]: It would be useful for Mac client as well :)

Comment (mzhurovich) [inline Blocks/HTTP/impl/posix_client.h]: And even more useful when we add extra parameters to cookies

Comment (dkorolev) [inline Blocks/HTTP/impl/posix_client.h]: :+1: da6a41206b005fe8425c4d44fcfa11948605d2a2

Comment (dkorolev) [inline Blocks/HTTP/impl/posix_client.h]: And :+1: d643dea

**Pull request merged**